docs/products/kafka/howto/schema-registry.rst

```diff
@@ -0,0 +1,156 @@
+         {"name": "referrer", "type": ["string", "null"], "default": "None"},
+         {"name": "ip", "type": ["string", "null"]}
+      ]
+    }
+
+The JSON configuration above defines a schema named ``ClickRecord`` in the namespace ``io.aiven.avro.example`` with fields ``session_id``, ``browser``, ``campaign``, ``channel``, ``referrer`` and ``ip`` and related data types. 
+
+Once the schema is defined, you need to compile it, and it can be done **manually** or **automatically**.
+
+Manual schema compilation
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In case of manual schema compilation, download ``avro-tools-1.11.0.jar`` from https://avro.apache.org/releases.html#Download or via maven using the following::
+
+    mvn org.apache.maven.plugins:maven-dependency-plugin:2.8:get -Dartifact=org.apache.avro:avro-tools:1.11.0:jar -Ddest=avro-tools-1.11.0.jar
+
+The schema defined in the previous step, can be now compiled to produce a Java class ``ClickRecord.java`` in the ``io.aiven.avro.example`` package (taken from the ``namespace`` parameter)::
+
+    java -jar avro-tools-1.11.0.jar compile schema ClickRecord.avsc .
+
+.. Note:: 
+
+    The package structure will also be generated if missing.
+
+Auto schema compilation
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+With auto, the schema is compiled during the project build with, for example, ``maven-avro-plugin`` or ``gradle-avro-plugin``.
+The following is a configuration example for ``maven-avro-plugin`` when ``ClickRecord.avsc`` is stored in the path ``src/main/avro/ClickRecord.avsc``::
+
+    <plugin>
+        <groupId>org.apache.avro</groupId>
+        <artifactId>avro-maven-plugin</artifactId>
+        <version>${avro-maven-plugin.version}</version>
+        <executions>
+            <execution>
+                <id>schemas</id>
+                <phase>generate-sources</phase>
+                <goals>
+                    <goal>schema</goal>
+                    <goal>protocol</goal>
+                    <goal>idl-protocol</goal>
+                </goals>
+                <configuration>
+                    <sourceDirectory>${project.basedir}/src/main/avro/</sourceDirectory>
+                    <outputDirectory>${project.basedir}/src/main/generated-sources/</outputDirectory>
+                </configuration>
+            </execution>
+        </executions>
+    </plugin>
+
+The automatically generated Avro-schema code will be stored under the folder ``${project.basedir}/src/main/generated-sources/``.
+
+Set consumer and producer properties for schema registry
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The full code to create consumer and producers using the Schema Registry in Aiven for Apache Kafka can be found in the `Aiven examples GitHub repository <https://github.com/aiven/aiven-examples/tree/master/solutions/kafka-schema-registry>`_. The following contains a list of the properties required.
+
+Once you implement producers and consumers on your own then
```

Comment:
My apologies, I think it was clearer without this line!  Since the code snippets and the aiven-example project are presented here at the same time and side-by-side, it's unnecessary.